client/components/phone-input/test/phone-number__misc.test.ts

```diff
@@ -0,0 +1,131 @@
+		{ input: [ '1', '2', '3', '4' ], index: 3, expected: 3 },
+		{ input: [ '1', '2', '3', '4' ], index: 4, expected: 4 },
+		{ input: [ '1', '2', '3', '4' ], index: 5, expected: 4 },
+		{ input: [ '1', 'x', '3', '4' ], index: 3, expected: 2 },
+		{ input: [ '1', 'x', '3', '4' ], index: 4, expected: 3 },
+		{ input: [ '1', 'x', '3', '4' ], index: 5, expected: 3 },
+		{ input: [ 'x', 'y', 'z', 'w' ], index: 3, expected: 0 },
+		{ input: [ 'x', 'y', 'z', 'w' ], index: 4, expected: 0 },
+		{ input: [ 'x', 'y', 'z', 'w' ], index: 5, expected: 0 },
+	] )( 'Returns correct cutoff index for input $input', function ( { input, index, expected } ) {
+		const result = numDigitsBeforeIndex( input, index );
+		expect( result ).toEqual( expected );
+	} );
+} );
+
+describe( 'getUpdatedCursorPosition', function () {
+	test.each( [
+		{ oldvalue: '', newvalue: '+', position: 1, expected: 1 },
+		{ oldvalue: '+5', newvalue: '+', position: 2, expected: 1 },
+		{ oldvalue: '555', newvalue: '+', position: 3, expected: 1 },
+		{ oldvalue: '23', newvalue: '234', position: 2, expected: 3 },
+		{ oldvalue: '234', newvalue: '234-5', position: 3, expected: 5 },
+		{ oldvalue: '223-2', newvalue: '223-22', position: 5, expected: 6 },
+		{ oldvalue: '223-2', newvalue: '223-232', position: 5, expected: 7 },
+		{ oldvalue: '223-2', newvalue: '223-22', position: 4, expected: 5 },
+		{ oldvalue: '223-2', newvalue: '223-22', position: 3, expected: 5 },
+		{ oldvalue: '223-25', newvalue: '223-225', position: 5, expected: 6 },
+		{ oldvalue: '223-25', newvalue: '223-2325', position: 5, expected: 7 },
+		{ oldvalue: '223-25', newvalue: '223-225', position: 4, expected: 5 },
+		{ oldvalue: '223-25', newvalue: '223-225', position: 3, expected: 5 },
+		{ oldvalue: '234-5', newvalue: '234', position: 5, expected: 3 },
+		{ oldvalue: '+1 234', newvalue: '+1 234-5', position: 6, expected: 8 },
+		{ oldvalue: '(802) 234', newvalue: '(802) 234-5', position: 9, expected: 11 },
+		{ oldvalue: '802-2343', newvalue: '(802) 234-39', position: 8, expected: 12 },
+		{ oldvalue: '(802) 234-39', newvalue: '802-2343', position: 12, expected: 8 },
+		{ oldvalue: '(802) 234-3943', newvalue: '(802) 233-943', position: 9, expected: 8 },
+		{ oldvalue: '(802) 687-6234', newvalue: '802-6234', position: 6, expected: 3 },
+		{ oldvalue: '(802) 687-6234', newvalue: '(802) 555-6234', position: 6, expected: 9 },
+		{ oldvalue: '124', newvalue: '1234', position: 2, expected: 3 },
+		{ oldvalue: '124', newvalue: '123-4', position: 2, expected: 3 },
+		{ oldvalue: '224', newvalue: '223-764', position: 2, expected: 6 },
+		{ oldvalue: '124-4', newvalue: '124-64', position: 4, expected: 5 },
+		{ oldvalue: '(802) 614-21', newvalue: '(802) 619-421', position: 8, expected: 9 },
+		{ oldvalue: '(802) 222-2222', newvalue: '6', position: 14, expected: 1 },
+		{ oldvalue: '+1 802 222-2222', newvalue: '6', position: 15, expected: 1 },
+	] )(
+		`Returns correct cursor position for inputs $oldvalue, $, newvalue`,
```

Comment:
I think something got messed up in the name here, I believe this should read...

`Returns correct cursor position for inputs $oldvalue, $newvalue`

Reply:
I was using regex to mass-edit the parametrized inputs and I believe that's why it got messed up.

Reply:
Addressed in e10751336012e40c886f517b08cc4f2cbe15d2e0.